build/ci/Makefile

```diff
@@ -171,6 +169,14 @@ ci-gke-cleanup: ci-e2e-delete-cluster
     	-e "GCLOUD_PROJECT=$(GCLOUD_PROJECT)" \
     	-e "GKE_CLUSTER_NAME=$(GKE_CLUSTER_NAME)" \
     	-e "GKE_SERVICE_ACCOUNT_KEY_FILE=$(GO_MOUNT_PATH)/build/ci/$(GKE_CREDS_FILE)" \
-    	cloud-on-k8s-ci-e2e \
+    	$(IMAGE) \
     	bash -c "GKE_CLUSTER_VERSION=1.11 $(GO_MOUNT_PATH)/operators/hack/gke-cluster.sh auth && \
     	 	$(GO_MOUNT_PATH)/build/ci/delete_unused_disks.py"
+
+# Build and push Docker image for CI
+.ONESHELL: #https://www.gnu.org/software/make/manual/html_node/One-Shell.html
```

Comment:
Should this be closer to the top of the file? I believe it will cause all targets to run in one shell, but it looks like only single target is affected.

Reply:
I needed it only for this particular target. Other targets don't need it. Therefore, my personal preference will be to keep it only for target which needs it. 
If there will be more targets which needs it, then we definitely should use for all targets.

Reply:
Do we really need `ONESHELL`? Is it just to be in the right directory to build the image?

Suggestions: 
`cd $(ROOT_DIR) && docker build -f build/ci/Dockerfile -t $(IMAGE) .` or
`docker build -f $(ROOT_DIR)/build/ci/Dockerfile -t $(IMAGE) $(ROOT_DIR)`

Sorry, I really like Makefile but I don't like to use some _exotic_ features.

Reply:
@artemnikitin what I meant is that I _think_ even if you put it there it affects other targets as well, so it might be more clear to put it higher up. From the linked docs:
> if the .ONESHELL special target appears anywhere in the makefile then all recipe lines for each target will be provided to a single invocation of the shell

This might be especially risky with secrets and @ use when someone assumes output won't be echo'ed, but it will.

Reply:
Fixed